### DISCUSSION TITLE
support for mTLS

### DISCUSSION THREAD

**Discussion by anju-geetha-bai-db** [decke/smtprelay#138] [Ideas]
Hi Team,
is it possible to add mTLS support?
we  need encrypted connection between relay and smtpGateway and the relay should be authenticated based on client certificate.
I  played around with the codebase , but unsuccessful .
The error is  " tls: first record does not look like a TLS handshake."

**Comment by decke** [comment]
If you want to really put this into smtprelay you will need to extend `getTLSConfig()`.

As an alternative you could use something like a proxy in front to handle mTLS
https://github.com/habibiefaried/mtls-tcp-proxy
